spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/Mqttv5PahoMessageHandler.java

```diff
@@ -165,11 +166,21 @@ protected void doStart() {
 			this.mqttClient.connect(this.connectionOptions).waitForCompletion(getCompletionTimeout());
 		}
 		catch (MqttException ex) {
-			ApplicationEventPublisher applicationEventPublisher = getApplicationEventPublisher();
-			if (applicationEventPublisher != null) {
-				applicationEventPublisher.publishEvent(new MqttConnectionFailedEvent(this, ex));
+			if (this.connectionOptions.isAutomaticReconnect()) {
```

Comment:
I'm not sure why we need to call `reconnect()`. For me the only purpose of calling `reconnect()` was to fix the automaticReconnect option. But maybe I've missed something. 
I've updated the code to connect in `handleMessageInternal()`. But the code probably needs to be improved in order to avoid spamming the MQTT broker with invalid connection (bad credentials, ... )

Reply:
I think that one is already out of our scope.
For that reason this Paho MQTT client raises this:
```
		if ((reasonCode == MqttClientException.REASON_CODE_FAILED_AUTHENTICATION) || 
			(reasonCode == MqttClientException.REASON_CODE_NOT_AUTHORIZED)) {
			return new MqttSecurityException(reasonCode);
		}
```
So, it's up to end user to catch this exception this or other way and deal with it respectively.